exploration/selection-matching-options.md

```diff
@@ -0,0 +1,547 @@
+when *   *   one {You have {$days} day to purchase {$items} items to earn {$coins} coin}
+when *   *   *   {You have {$days} days to purchase {$items} items to earn {$coins} coins}
+```
+
+Let's use values `0`/`3`/`3`. The first _selector_, `{$days :plural}`, matches the value `0` and the `*` rule, producing this matrix:
+
+```
+0   *   *
+*   one one
+*   one *
+*   *   one
+*   1   *
+*   *   *
+```
+
+The second _selector_ `{$items :plural}`, matches the `*` rule, producing this matrix:
+
+```
+0   *   *
+*   *   one
+*   *   *
+```
+
+The third and final _selector_ `{$coins :plural}` matches the `*` rule, producing this matrix:
+
+```
+0   *   * <-- winner
+*   *   *
+```
+
+The value `3`/`1`/`3` produce different matching. The first selector now only matches the `*` rule:
+
+```
+*   one one
+*   one *
+*   *   one
+*   1   *
+*   *   *
+```
+
+The second _selector_ matches the explicit value `1`, which is prefers to the keyword value `one` and the default value `*` _in that order_:
+
+```
+*   1   *
+*   one one
+*   one *
+*   *   one
+*   *   *
+```
+
+The final _selector_ matches the default value `*` but not the keyword `one`, thus producing this matrix:
+
+
+```
+*   1   * <-- winner
+*   one *
+*   *   *
+```
+
+### Column-First with Required `*`
+
+This is a "sorted" best match algorithm that works as follows: each _selector_ provides a "comparator" for values in its column (such as computing a weight for the value in its column). Rows that contain a non-matching value for any selector are eliminated as potential matches. The default value `*` always matches. Ordering is maintained for preceding columns, that is, _selector_ number 2 can only reorder items whose _selector_ number 1 key match. The highest ranking _key_ is returned as the _pattern_. Ties are broken by column. If no matching row is found, returns an error.
+
+
+**Pros**
++ Variants can be written in any order and produce a consistent result.
++ Selector developers can write complex matches that produce different quality matches for the same value. For example, `{|1| :plural}` matches both the variant `1` and the variant `one`, but prefers the value `1`. The plural _selector_ does not need to communicate with the other _selectors_ in order to arrive at the best matching pattern.
++ Translators do not need to worry about the order of variants or need to reorder variants (which can be difficult to do when only the translation segment for the pattern is shown or when only a changed or generated _variant_ is exposed to translation.
++ Translation tools do not have to preserve the order of _variants_ and are free to send only the translatable segment (the pattern) for translation.
++ Easier to evaluate visually than sorting strategies.
+
+**Cons**
+- Developers cannot override the order that the _selector_ provides unless this is exposed as a feature of the given _selector_.
+- Can require more processing than First-Match
+
+
+### Column-First with Optional `*`
```

Comment:
The question of whether the `*` is required or optional seems related, but orthogonal. I think it's a valid question for all strategies presented. Perhaps move it to its own section? (In this PR or later.)